### ISSUE TITLE
🤔 feedback needed — vf-forms refactor and restyling 

### ISSUE THREAD

**Issue by sturobson** [visual-framework/vf-core#1229]
There have been a few questions raised about the styling of some of the form elements, specifically:

- their interactive states
- their colour contrast
- their spacing

Here's a before/after of where #1228 is at

*new radio and checkbox designs*
![Screenshot 2020-11-09 at 12 57 13](https://user-images.githubusercontent.com/925197/98825236-2b62fa80-242c-11eb-8a3d-4ca3df0b30df.png)

*new radio and checkbox interactive states*
![Screenshot 2020-11-09 at 13 06 11](https://user-images.githubusercontent.com/925197/98825308-3cac0700-242c-11eb-9007-3ce7873dfad6.png)
![Screenshot 2020-11-09 at 13 06 24](https://user-images.githubusercontent.com/925197/98825327-42095180-242c-11eb-8900-dcc2901e2448.png)

*before*
![Screen Shot 2020-11-11 at 14 11 10](https://user-images.githubusercontent.com/925197/98821780-e5a43300-2427-11eb-93a0-dca2c5674ef8.png)

*after*
![Screen Shot 2020-11-11 at 14 09 59](https://user-images.githubusercontent.com/925197/98821829-f5bc1280-2427-11eb-991b-8d4c2438d2d9.png)

+ @nikiforosk / @gabsie

**Comment by sturobson** [comment]
- the placeholder text might need looking at for colour contrast still. 
- the inline `vf-form__checkbox` might need fixing up still.

**Comment by nikiforosk** [comment]
Thanks for walking us through this @sturobson  Below are a few thoughts with some supporting links:

1. I feel that the darker border makes the text box easier to spot. Maybe the mouseover effect could be a form of highlight [like Google](https://www.google.com/) does? 

2. Form design gury Caroline Jarrett strongly recommends against putting labels (such as hint or placeholder text) in the text box:
https://www.effortmark.co.uk/dont-put-labels-inside-text-boxes/

As Joseph mentioned, in the Account manager they put the hint text below the primary label and the warning text under the text box:

![example-form-account-manager](https://user-images.githubusercontent.com/20280139/98959011-c88c6480-24fa-11eb-99c0-9dfb35712fac.png)

Google forms also places the warning text under the text box (or the radio buttons, etc):

![Screenshot 2020-11-12 at 15 13 53](https://user-images.githubusercontent.com/20280139/98959494-3d5f9e80-24fb-11eb-855b-893fa0984081.png)

3. Mouseover effect on radiobuttons and tickboxes

I feel that the filling in the radiobutton or tickbox on mouseover may actually mislead users into thinking that they have selected the option upon mouseover. 

Google forms uses a subtle highlight:
![Screenshot 2020-11-12 at 15 27 13](https://user-images.githubusercontent.com/20280139/98959832-98919100-24fb-11eb-8153-eaa91873878e.png)

This may be easier to view in this test form:
https://docs.google.com/forms/d/e/1FAIpQLSddiznU6qxSe-WgHbzHXrVUhjGZuMa3G_ry-f0Am6T6Mg3eEg/viewform

Carla made a review of the design rationale of the forms in the account manager so we should include her in this discussion:
https://docs.google.com/document/d/1qRwNn1TUWEyIniW_D7oArHrNs37-BDDTvtxFBCufdik/edit?usp=sharing

I hope this is helpful!

**Comment by gabsie** [comment]
Hey, thanks, @sturobson and everyone

Yes from me, too on our decision earlier - error text at the bottom and with how this was implemented for the Acct manager  - hint text under the label, error text under the form.

I think I would also want to see it in action, because then we can additionally give feedback when it's interactive!

I really liked our session today, more of those!
Gabs

**Comment by sturobson** [comment]
Thanks Niki, some replies below:

> Maybe the mouseover effect could be a form of highlight like Google does?

This might work, as we have a drop-shadow effect on our card components (if they're a link), so it's something that already exists in the visual framework. 

> As Joseph mentioned, in the Account manager they put the hint text below the primary label and the warning text under the text box:

here's a 'bells and whistles' example — 
![Screenshot 2020-11-12 at 15 58 44](https://user-images.githubusercontent.com/925197/98963864-08a21600-2500-11eb-9612-0ff2e039f2ab.png)

the icon is taken from the streamlines icon set that the VF2.x is using. There is also a 'filled' variant.

**Comment by cindyebi** [comment]
Hi guys, @sturobson and I discussed the checkbox variants yesterday and I have some comments. The **current** checbox variants are as follows: 

Default:
<img width="199" alt="default" src="https://user-images.githubusercontent.com/75834903/108526144-b45dbd80-72c8-11eb-81ad-4b6d979d91c9.png">

Ticked
<img width="211" alt="checked" src="https://user-images.githubusercontent.com/75834903/108526140-b3c52700-72c8-11eb-9fd5-1c3662afd6f7.png">

Hover
<img width="211" alt="hover" src="https://user-images.githubusercontent.com/75834903/108526148-b45dbd80-72c8-11eb-9dfc-de14cd525ab0.png">

I noticed the checkboxes are currently transparent, without white background inside of the box when hovered and ticked. I think these could be consistent and I'd like to also propose a slightly different style for the **hover** and **ticked** variants:
<img width="592" alt="Screenshot 2021-02-19 at 15 42 19" src="https://user-images.githubusercontent.com/75834903/108526441-0ef71980-72c9-11eb-965f-659f46b8b8fb.png">
Basically, the Hover state has the same behavior as the Hover Form field: white box with black outline and drop shadow. The ticked state has a blue filled box with a white tick icon to keep it visually consistent with the radio buttons. 

Thoughts?

**Comment by gabsie** [comment]
Hey @cindyebi 

I like the blue-filled checkbox, but it might be good to actually see these in action, i.e. in a real form.. in order to have a better view and compare them..  :)

Hope that's not too much trouble
Gabs

**Comment by cindyebi** [comment]
Actually this was inspired by the Covid19dataportal page: https://www.covid19dataportal.org/host-sequences?db=human-experiment&size=15&facets=TAXONOMY:9606#search-content 
<img width="253" alt="Screenshot 2021-02-19 at 16 33 54" src="https://user-images.githubusercontent.com/75834903/108532969-5af98c80-72d0-11eb-9bc4-a7d6aab490a7.png">

I wonder if @khawkins98 had any thoughts about this

**Comment by sturobson** [comment]
as there's a space of white around radio buttons. I've got a branch working looking similar.

![Screenshot 2021-02-19 at 16 41 30](https://user-images.githubusercontent.com/925197/108533961-8761d880-72d1-11eb-94b0-e06a3ed4b996.png)

**Comment by cindyebi** [comment]
> as there's a space of white around radio buttons. I've got a branch working looking similar.
> 
> ![Screenshot 2021-02-19 at 16 41 30](https://user-images.githubusercontent.com/925197/108533961-8761d880-72d1-11eb-94b0-e06a3ed4b996.png)

Looks great, will we have the checkboxes in white fill when hovered too (currently transparent)?

**Comment by nikiforosk** [comment]
My two cent on this is (or are?):

1. The gray border of the unselected box may be hard to see (especially in low resolution screens or if the background is grayish too). I experienced this problem myself when I was looking at the new embl jobs pages earlier this year (before I switched to a better external monitor):

![Screenshot 2020-06-03 at 13 24 54](https://user-images.githubusercontent.com/20280139/108732769-c7b99480-7525-11eb-8e99-0aa185fc4190.png)

I thicker black border will be preferable in my opinion to overcome this issue (as I mentioned previously in this thread about the border of the actual text box). 

I generally find the behaviour of the check boxes in gov.uk quite straightforward:

https://design-system.service.gov.uk/components/checkboxes/

And in action:
https://www.gov.uk/search/all?keywords=passports&content_purpose_supergroup%5B%5D=research_and_statistics&content_purpose_supergroup%5B%5D=transparency&order=relevance

<img width="374" alt="Screenshot 2021-02-22 at 15 42 14" src="https://user-images.githubusercontent.com/20280139/108733582-8c6b9580-7526-11eb-81cc-e2eb67646e1a.png">

They use a gray highlight around the box for the hover state, which again is relatively easy to spot (at least on a white background). 

They also use a black tick on a white background once the box is ticked (with a yellow added border which disappears when you click somewhere else on the page). Again to me this looks easier to spot than a white tick in a blue background.

2. In the training site, we have the additional requirement that [when you select the check boxes of one category (e.g. on-demand training) the check boxes of the other category (e.g. live training) should look un-selectable](https://www.ebi.ac.uk/training/search-results?query=bioinformatics&domain=ebiweb_training_online&page=1&facets=duration:Less%20than%201%20hour). 

These un-selectable boxes are currently shown as dark gray filled boxes but we'd probably like to have something more elegant:
 
![Screenshot 2021-02-22 at 16 01 24](https://user-images.githubusercontent.com/20280139/108734298-3fd48a00-7527-11eb-9acd-e3732d983264.png)

In any case, we need to add "un-selectable" to the existing statuses to account for this situation in the actual pattern.

3. I feel that the consistency with the radio buttons is less important. But if we wanted to achieve that, I'd use a black fill for the radio button, again like gov.uk does:

https://design-system.service.gov.uk/components/radios/

[When you see this pattern in action](https://www.passport.service.gov.uk/filter/overseas?_ga=2.179660326.308435545.1614008705-599809005.1614008705), it seems to lack a mouseover effect (but it has a yellow border once selected and until you click away).

![Screenshot 2021-02-22 at 16 10 01](https://user-images.githubusercontent.com/20280139/108735591-a73f0980-7528-11eb-9182-0ff9a5608666.png)

I hope this is helpful.

**Comment by sturobson** [comment]
replies:

> The gray border of the unselected box may be hard to see (especially in low resolution screens or if the background is grayish too)

I guess the question arises to why is the background a different colour, and where has this been decided from within the VF. 

I have no issue with moving away from the grey to either a darker grey or the black. Although it does raise a question as to why the `vf-card` hasn't got a black border etc etc. 

> I generally find the behaviour of the check boxes in gov.uk quite straightforward:

I don't get any hover interactions at all here, using Safari. 

> These un-selectable boxes are currently shown as dark gray filled boxes but we'd probably like to have something more elegant:

I think I agreed that the `disabled` variants for checkboxes could a little nicer on the eye (tbd).

> I feel that the consistency with the radio buttons is less important. 

Taken the govuk as an example - there's consistency here as the checkbox has a black tick and the radio disc is black. 

We use blue here as:
a) it's the corporate blue
b) the default browser UI is blue (which is often changed to match the brand - like govuk do here with it being black)

**Comment by cindyebi** [comment]
@nikiforosk @sturobson and I discussed these form states this morning and we propose these solutions:
![Screenshot 2021-02-22 at 16 29 16](https://user-images.githubusercontent.com/75834903/108738146-3816e480-752b-11eb-9f87-5f26c3d503b5.png)
![Screenshot 2021-02-22 at 16 29 23](https://user-images.githubusercontent.com/75834903/108738151-38af7b00-752b-11eb-90d4-c14cdebe20fe.png)

I agree we need better "disabled" checkbox and radio button states but we want to keep the visuals coherent throughout the components. In this case, blue is our interactive colour so I think it makes sense to use Blue instead of black. We also want to stay away from the shadow effect when hovered or focused, since it doesn't have a good contrast so we proposed a thicker black outline for the Hover and Focus states.

**Comment by nikiforosk** [comment]
This is fine but as @sturobson also mentioned also we need a disabled variant.

**Comment by gabsie** [comment]
Hi, everyone
Happy with suggestions and how this is progressing. My only comment at this stage is that purely visually the ticked states look smaller than the unticked ones (I know that's some kind of illusion). That's why I wanted to see them in a real form with real content.. so that they don't 'look' inconsistent.

I guess we can have a very short discussion this afternoon. :)
Gabs

**Comment by sturobson** [comment]
after yesterdays VF/UX Chat:

https://user-images.githubusercontent.com/925197/108980619-c3a38900-7683-11eb-9250-8a38fe57f661.mov

**Comment by khawkins98** [comment]
Looking good.

**Comment by gabsie** [comment]
I really like how these turned out. Is there a way to avoid the slight white on hover of the radio buttons!
Can't wait to use those in my forms soon.
Gabs

**Comment by sturobson** [comment]
Hey @gabsie that slight white crinkle around the blue on hover is a rendering issue with Firefox :( looks fine in Chrome and Safari.

**Comment by sturobson** [comment]
A few more tweaks to forms that you all might like to comment on:

### Changing the `<select>` element icon
Currently this icon is something taken from the original sketch files. As we're moving to use the Streamline Icons for things. I've updated the icon. After discussion with @cindyebi and @nikiforosk on wether to match the native browser UI (up and down arrows) or something else. We've decided on the down arrow only (as per GOVUK).

![Screenshot 2021-02-24 at 12 58 33](https://user-images.githubusercontent.com/925197/109004580-d5475980-76a0-11eb-8dd5-35773bcf04aa.png)

### Updating the search input 'reset' button icon.
Some browsers offer a UI 'x' to reset the search input. Currently we have a larger `x` than the native browser UI because of feedback on an EBI project. I've now updated this so it makes use of the streamline icon equivalent. 
(note: the border is darker here than the `<select>` image because I was focused on the search input at the time I took a screenshot)

<img width="608" alt="Screenshot 2021-02-24 at 12 59 19" src="https://user-images.githubusercontent.com/925197/109004718-0a53ac00-76a1-11eb-8ffa-7dcc4707dc6e.png">

### Checkbox 'tick'
The current 'tick' is quite a thick stroked icon. It is from the streamline icon set but I have found a 'tick' that matches the strokes of the `vf-tree` and the new `vf-form__select` component.

Old:  
![Screenshot 2021-02-24 at 12 58 03](https://user-images.githubusercontent.com/925197/109004883-396a1d80-76a1-11eb-9d8e-cb95ea8ebc5f.png)

New:

![Screenshot 2021-02-24 at 12 58 13](https://user-images.githubusercontent.com/925197/109004902-40912b80-76a1-11eb-9627-654c0bd26d8e.png)

**Comment by cindyebi** [comment]
@sturobson I suggest we use this X icon instead to match the outline thickness of the other icons we use. Also downloaded from Streamline icon. 

![Screenshot 2021-02-26 at 10 20 21](https://user-images.githubusercontent.com/75834903/109288324-9b02c700-781c-11eb-9c4f-c24da71d609c.png)


![streamline-icon-interface-delete-1@140x140](https://user-images.githubusercontent.com/75834903/109288296-8faf9b80-781c-11eb-9a8f-503fa71172f5.png)

**Comment by sturobson** [comment]
Nice, yeah the thickness of the proposed matches the other icons - so I'll get this into #1390

**Comment by sturobson** [comment]
![Screenshot 2021-03-02 at 09 34 21](https://user-images.githubusercontent.com/925197/109628372-969b1e80-7b3a-11eb-8260-ad732f3a2474.png)
![Screenshot 2021-03-02 at 09 35 23](https://user-images.githubusercontent.com/925197/109628466-af0b3900-7b3a-11eb-830f-7d0e872edfee.png)


I've had another look at the 'disabled' versions of the checkbox and the radio button and tidied them up so they meet some WCAG accessibility guidelines.

**Comment by cindyebi** [comment]
> ![Screenshot 2021-03-02 at 09 34 21](https://user-images.githubusercontent.com/925197/109628372-969b1e80-7b3a-11eb-8260-ad732f3a2474.png)
> ![Screenshot 2021-03-02 at 09 35 23](https://user-images.githubusercontent.com/925197/109628466-af0b3900-7b3a-11eb-830f-7d0e872edfee.png)
> 
> I've had another look at the 'disabled' versions of the checkbox and the radio button and tidied them up so they meet some WCAG accessibility guidelines.

Looks good, though I would suggest keeping the disabled checkbox empty without the tick icon inside to show that it is disabled?

**Comment by sturobson** [comment]
updated: 

![Screenshot 2021-03-09 at 10 06 45](https://user-images.githubusercontent.com/925197/110454558-4a625800-80bf-11eb-8220-6ddb429f0f59.png)

+ @cindyebi
